Automattic/simplenote-ios
# Issue/1407 reference count incorrect

### Fix
This PR updates the reference count logic in the Note Information sheet to calculate the number of internal references are contained in a linked note.  This fixes #1147 

Previous:
<img width="400px:" src="https://user-images.githubusercontent.com/73365754/107752872-febec780-6d27-11eb-8318-63887d50a700.png">

New:
<img src="https://cdn-std.droplr.net/files/acc_593859/6yXz3m"/> 


### Test
***(Required)*** List the steps to test the behavior.  For example:
1. Create a note to be the note that is referenced to (NoteA).
2. In the new note, go to the note menu > Get internal link
3. go to another note (NoteB) and paste in the internal link one time into the note
4. Go to a third note (NoteC) and paste the internal link two different times in the content of the note
5. Return to NoteA and open the information sheet

There you should see both NoteB and NoteC listed as references, noteB should say 1 reference noteC should say 2

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
> These changes do not require release notes.
